src/erl_optics_foil_server.erl

```diff
@@ -0,0 +1,115 @@
+-module(erl_optics_foil_server).
+-include("erl_optics.hrl").
+
+-behaviour(gen_server).
+
+-export([
+    init/1,
+    handle_cast/2,
+    handle_call/3,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-export([
+    start_link/0,
+    add_lens/2,
+    stop/0
+]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {
+    foil_namespace :: ?NS,
+    new_lenses  :: map()
+}).
+
+%%%=========
+%%% API
+%%%=========
+
+%Modes: carbon | blank
+
+start_link() ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+stop() ->
+    gen_server:cast(?SERVER, stop).
+
+%%%==========
+%%% Callbacks
+%%%==========
+
+%% TODO: Change the timer to start after a new lens has been
+%%       declared and stop after a while to avoid a permanent
+%%       server doing useless things.
+
+init(_) ->
+    Interval = ?ENV(?ENV_FOIL_RELOAD_INTERVAL, ?DEFAULT_FOIL_RELOAD_INTERVAL),
+    timer:send_interval(Interval, reload_foil),
```

Comment:
You should store the `timer_ref()` in the server state so you can cancel if needed.